### PR TITLE
DVC-8004 log + code cleanup

### PR DESCRIPTION
**Pull request by chris-hoefgen**
- standardize on f-strings for formatting
- cleaned up some of the logger usage, downgrading common messages to lower tiers
- more code comments

### CODE REVIEW

File: devcycle_python_sdk/cloud_client.py [hunk 1]
```diff
@@ -61,15 +73,15 @@ def variable(self, user: DevCycleUser, key: str, default_value: Any) -> Variable
         try:
             variable = self.bucketing_api.variable(key, user)
         except CloudClientUnauthorizedError as e:
-            logger.warning("DevCycle: SDK key is invalid, unable to make cloud request")
+            logger.warning("DevCycle SDK key is invalid, unable to make cloud request")
             raise e
         except NotFoundError:
-            logger.warning("DevCycle: variable not found: %s", key)
```
Comment (JamieSinn) [inline devcycle_python_sdk/cloud_client.py:67]: Do we have a prefix handler setup for this to show devcycle as the source of the log?

File: devcycle_python_sdk/local_client.py [hunk 2]
```diff
@@ -103,18 +113,27 @@ def variable(self, user: DevCycleUser, key: str, default_value: Any) -> Variable
             if variable:
                 return variable
         except VariableTypeMismatchError:
-            logger.info("Variable type mismatch, returning default value")
+            logger.debug("Variable type mismatch, returning default value")
         except Exception as e:
-            logger.error("Error retrieving variable for user: %s", e)
+            logger.warning(f"Error retrieving variable for user: {e}")
```
Comment (suthar26) [inline devcycle_python_sdk/local_client.py:118]: Do we want all of our exceptions to be the same logger level? there are some with logger.error and others with logger.warning

File: devcycle_python_sdk/cloud_client.py (hunk 1 above)
Comment (chris-hoefgen) [reply devcycle_python_sdk/cloud_client.py:67]: Well all of the loggers do have names that should include the path so developers should be able to filter and control the log output based on the logger names

**Review by suthar26 — APPROVED** [review]
overall changes look good, if the mismatch in some logger levels is intended then its good to merge

File: devcycle_python_sdk/local_client.py (hunk 2 above)
Comment (chris-hoefgen) [reply devcycle_python_sdk/local_client.py:118]: yeah, i kind of reserved error for when things go bad inside the SDK and the state changed, such as if we start getting a Forbidden from the config CDN and stop the polling thread, as opposed to general network exceptions which will be retried.

**Pull request merged**